test/cpp/jit/test_misc.cpp

```diff
@@ -1992,6 +1992,66 @@ TEST(ProfilerTest, Basic) {
   checkShape(tanh_n->inputs().at(0)->node()->ty(attr::profiled_type), eltwise);
 }
 
+TEST(ProfilerTest, OptionalProfiling) {
```

Comment:
@Krovatkin  I think this is actually testing the case where sometimes the tensor is provided, and sometimes it is `None` (which is new logic, since before we didn't have optional tensors